src/LiveComponent/doc/index.rst

```diff
@@ -2389,6 +2384,50 @@ For example, if you declare the following bindings::
 And you only set the ``query`` value, then your URL will be updated to
 ``https://my.domain/search?query=my+query+string&mode=fulltext``.
 
+Controlling the Query Parameter Name
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Instead of using the prop's field name as the query parameter name, you can use the ``as`` option in your ``LiveProp``
+definition::
+
+    // ...
+    use Symfony\UX\LiveComponent\Metadata\UrlMapping;
+
+    #[AsLiveComponent]
+    class SearchModule
+    {
+        #[LiveProp(writable: true, url: new UrlMapping(as: 'q')]
+        public string $query = '';
+
+        // ...
+    }
+
+Then the ``query`` value will appear in the URL like ``https://my.domain/search?q=my+query+string``.
+
+If you need to use a custom parameter name on a specific page, you can override the mapping on a per-template basis:
+
+.. code-block:: html+twig
+
+    {# Pass a mapping object #}
+    <twig:SearchModule :data-live-url-mapping-query="{ alias: 'q'}" />
+
+    {# Or configure the mapping option directly #}
+    <twig:SearchModule data-live-url-mapping-query-alias="q" />
```

Comment:
Let's talk about this approach :). What I don't love about it is that users need to know about more `data` attribute options. And we need to make sure we keep this value on the frontend so we can reference it on re-renders.

The Livewire solution is to allow the mapping to be done in a method if you have something custom: https://livewire.laravel.com/docs/url#using-the-querystring-method

We could even go more generic: adding some way of configure a `LiveProp` entirely:

```php
#[AsLiveComponent]
class SearchModule
{
    #[LiveProp(writable: true, 'modifier' => 'modifyQueryProp')]
    public string $query = '';

    #[LiveProp]
    public string $alias = null;

    public static function modifyQueryProp(LiveProp $prop): void
    {
        if (!$alias) {
             return;
        }

        $prop->url = new UrlMapping(as: $this->alias);
    }
}
```

Usage:

```php
<twig:SearchModule alias="q" />
```

A tiny bit more work for the user, but I think it's more understandable (and will also be more flexible). WDYT?

Reply:
Oh I like it! It looks even easier to set up. I have a few questions in mind: 
- Does the `alias` prop need to be a `LiveProp`?
- We need to make `LiveProp` writable in some way, is there any drawback?
- The `modifier` method will need the component to be mounted before it's called, so we can use data passed from templates. Are we sure all `LiveProp` options are safe to be overridden here?

Reply:
Thanks for the investigations! So now this "side feature" of the alias option is growing up. I would remove it from this PR and work on it in a dedicated one, where we could specifically address the `ChildComponentPartialRenderer` modifications you mentioned above. Wdyt?

Reply:
Ha! Indeed it is - it's exposed a bigger thing!

>  I would remove it from this PR and work on it in a dedicated one, where we could specifically address the ChildComponentPartialRenderer modifications you mentioned above. Wdyt?

It'd be great to focus on that part in a smaller PR. But as LiveComponents will become stable at the end of Feb, I don't want to merge this now, if it'll change later.

If you'd like, there could be a compromise: perhaps you create a PR now with this refactoring of the `getMetadata()` method and `ChildComponentPartialRenderer`? Then we get that merged and *then* update and merge this PR.

Thanks for your patient work - you PR's have proved to be complex in a good way: handling important things in the correct way.

Reply:
Alright, let's do this! 

> Thanks for your patient work - you PR's have proved to be complex in a good way: handling important things in the correct way.

Thank you!